### PR TITLE
fix: handle item iri with identifiers in LegacyIriConverter

### PR DESCRIPTION
**Pull request by jonnyeom**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Tickets       | #5620 
| License       | MIT

Hello, a small fix for when using the `\ApiPlatform\Core\Api\LegacyIriConverter`

## The Problem:
#5630 Introduced a nice way to fix the IriConverterInterface deprecation before turning off the `metadata_backward_compatibility_layer`. 

One issue I ran into is that `LegacyIriConverter::getIriFromResource` decides if the iri will be Item level or Collection level solely based on if the given `$item` parameter is a string vs an object.

With this methodology, I would have instantiate a new object just to trigger the Item level iri conversion.
(To compare, The new IriConverterInterface handles this decision smartly, using the metadata of the Operation and the given Object).

## Proposed solution
Previously, the old IriConverterInterface had 2 distinct methods to split this (`getItemIriFromResourceClass` and `getIriFromResourceClass`)— the only difference being an extra parameter `array $identifiers`.

This adds back the ability to identify a Resource based on className + identifiers.

### CODE REVIEW

File: src/Core/Api/LegacyIriConverter.php
Comment (dunglas) [inline src/Core/Api/LegacyIriConverter.php:68]:
```suggestion
            if ($identifiers = ($context['uri_variables'] ?? null)) {
```

**Review by dunglas — CHANGES_REQUESTED** [review]
Nitpicking, otherwise LGTM, but I'll wait for @soyuka review before merging because he knows this part better!

**Review by soyuka — APPROVED** [review]
lgtm

**Pull request merged**
